org-formation/650-identity-providers/_tasks.yaml

```diff
@@ -812,9 +810,8 @@ GithubOidcNbConvertDeploy:
     ProviderRoleName: !Sub ${resourcePrefix}-${appName}-nbconvert-deploy
     MaxSessionDuration: 7200
     ManagedPolicyArns:
-      - "arn:aws:iam::aws:policy/AmazonEC2ContainerRegistryFullAccess"
-      - "arn:aws:iam::aws:policy/AmazonLambdaFullAccess"
-      - "arn:aws:iam::aws:policy/CloudWatchLogsFullAccess"
+      - "arn:aws:iam::aws:policy/AdministratorAccess"
```

Comment:
hmm, In this case where you know this OIDC only needs to push to AWS ECR then you should probably scope it to just allow that.  I like the previously scoped permissions set you had better.

Reply:
OK but this may gonna take a few iterations. I know I need at least IAM to modify some policies/statements so I'll add that.

Reply:
> OK but this may gonna take a few iterations. I know I need at least IAM to modify some policies/statements so I'll add that.

The title of this PR is `Add oidc perms to push image to repo` so I thought that's all this was needed for?  if not, what else do you plan to deploy?

Reply:
ahh so you are actually deploying a lambda not just pushing to ECR?  Depending on the lambda you'll probably want admin because the lambda can setup a bunch of AWS resources.  

I'm just wondering why you chose to run lambda in docker instead of deploying just a bare lambda?  it seems like an over complicated solution unless there's a good reason.  i think this is a good comparison of the two approaches. https://www.knowbe4.com/careers/blogs/engineering/containerized-lambdas-vs-binary-pros-and-conscon

Reply:
I tried using the python runtime but the python library we're using is missing a binary dependency so I moved to docker so I was able to add the dependency... 
I looked at the synthesized template and see a role, a function and a function permission (so IAM and Lambda) but as you say the deployer role could/should need more permissions (hence best guess about additional ones).
Best practice is nice but they sure don't make it easy!
So back to admin or we try to at least restrict the services?

Reply:
@zaro0508 Thank you for your suggestion.  I am happy to make changes in support of the principle of least access. 
 But, why do you say that `GithubOidcSageBionetworksSynapseDockerRegistry` "only needs to push to AWS ECR"?  E.g., it needs to access EC2 (incl. ALB), Route53, Certificate Manager, Secrets Manager, tagging.  We would need some methodology to determine what permissions a stack deployment actually needs.

Reply:
sorry, my mistake @brucehoff.  The name of the resource threw me off, it made it sound like this OIDC was just for deploying stuff to ECR.  I see it's for Sage-Bionetworks/synapse-docker-registry which is doing a lot more.